src/Filesystem/Mime.php

```diff
@@ -270,36 +270,34 @@ public static function toExtension(string|null $mime = null): string|false
 	/**
 	 * Returns all available extensions for a given MIME type
 	 */
-	public static function toExtensions(string|null $mime = null, bool $matchWildcards = false): array
-	{
-		$extensions = [];
-		$testMime = fn (string $v) => static::matches($v, $mime);
-
-		foreach (static::$types as $key => $value) {
-			if (is_array($value) === true) {
-				if ($matchWildcards === true) {
-					if (A::some($value, $testMime)) {
-						$extensions[] = $key;
-					}
-				} else {
-					if (in_array($mime, $value) === true) {
-						$extensions[] = $key;
-					}
-				}
-			} else {
-				if ($matchWildcards === true) {
-					if ($testMime($value) === true) {
-						$extensions[] = $key;
-					}
-				} else {
-					if ($value === $mime) {
-						$extensions[] = $key;
-					}
+	public static function toExtensions(
+		string|null $mime = null,
+		bool $matchWildcards = false
+	): array {
+		// get all extensions
+		$extensions = array_keys(static::$types);
+
+		// filter extensions for given MIME type
+		$extensions = A::filter(
+			$extensions,
+			function ($extension) use ($mime, $matchWildcards) {
+				// get corresponding MIME types as array
+				$mimes = A::wrap(static::$types[$extension]);
+
+				if ($matchWildcards) {
```

Comment:
```suggestion
				if ($matchWildcards === true) {
```

Reply:
I was thinking that with PHP type hints added, we. could start to drop some of those explicit comparisons (which we once in K2 -> K3 started to use as ambigious values in K2 caused problems). I wouldn't do it everywhere, but maybe in places where the variable/method return is ensured to be boolean and also from reading the code still very clear what's happening.

What do you think?

Reply:
Interesting idea. I agree that with a `bool` type the same behavior is guaranteed with or without an explicit comparison, so it's not needed. It's probably more a case of getting used to it.

Reply:
I'll add it here to be consistent with the status quo. But maybe something to briefly discuss in our next call and agree if we would like to do this more going forward.